src/lib/message/L1ToL2Message.ts

```diff
@@ -259,6 +283,68 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     return null
   }
 
+  private async getFirstRedeem(
+    retryableCreationReceipt?: TransactionReceipt,
+    includeFail?: boolean
+  ): Promise<TransactionReceipt | null> {
+    includeFail = includeFail || false
+    const l2Network = await getL2Network(this.l2Provider)
+    const creationReceipt =
+      retryableCreationReceipt ||
+      (await this.l2Provider.getTransactionReceipt(this.retryableCreationId))
+    const autoRedeem = await this.getAutoRedeem(creationReceipt)
+    if (autoRedeem && (includeFail || autoRedeem.status === 1)) return autoRedeem
+    if (creationReceipt){
+      const creationBlockNumber = creationReceipt.blockNumber
+      const ticketId = creationReceipt.transactionHash
+      const iFace = new Interface([
+        'event RedeemScheduled(     bytes32 indexed ticketId,     bytes32 indexed retryTxHash,     uint64 indexed sequenceNum,     uint64 donatedGas,     address gasDonor )',
+      ])
+      const redeemTopic = iFace.getEventTopic('RedeemScheduled')
+
+      let redeemEventLogs: ethers.providers.Log[] = []
+      let fromBlockNumber: number = creationBlockNumber
+      let fromBlock = await this.l2Provider.getBlock(fromBlockNumber)
+      let increment: number = 1000
+      const creationBlock = await this.l2Provider.getBlock(creationBlockNumber)
+      const maxBlock = await this.l2Provider.getBlockNumber()
+      while(fromBlockNumber <= maxBlock) {
+        const toBlockNumber = Math.min(fromBlockNumber + increment, maxBlock)
+        redeemEventLogs = await this.l2Provider.getLogs( {fromBlock: fromBlockNumber, toBlock: toBlockNumber, topics: [redeemTopic, ticketId]})
+        if (redeemEventLogs.length != 0) {
+          const redeemEvents = redeemEventLogs.map(
+            r =>
+              (iFace.parseLog(r).args as unknown) as {
+                ticketId: string
+                retryTxHash: string
+                sequenceNum: BigNumber
+                donatedGas: BigNumber
+                gasDonor: string
+              }
+          )
+          if(includeFail) {
+            return await this.l2Provider.getTransactionReceipt(redeemEvents[0].retryTxHash)
+          }
+          const successfulRedeem = (await Promise.all(redeemEvents.map(
+            async e => await this.l2Provider.getTransactionReceipt(e.retryTxHash)))).filter(
+              r => r.status === 1)
+          if (successfulRedeem.length > 1) throw new Error("More than 1 successful redeem?")
```

Comment:
Should throw an `ArbTsError`. Should probably make the error message more descriptive too:
`Unexpected number of successful redeems. Expected only one redeem for transaction ${transactionHash}, but found ${successfulRedeem.length}.`